openedx/openedx-platform
# feat: Tag List on Unit page [FC-0036]

## Description

This adds:
- a new container subview to show the list of tags of a unit. 
- the "Manage tags" menu item on the component menu.
Both are under the `new_studio_mfe.use_tagging_taxonomy_list_page` flag.

![image](https://github.com/openedx/edx-platform/assets/11827305/83356cb6-1be6-49d8-9c5e-8f0bd6e28a31)
![image](https://github.com/openedx/edx-platform/assets/11827305/a2e78595-3e58-47ef-8890-64629bf15311)


## Supporting information

- Internal ticket: [FAL-3527](https://tasks.opencraft.com/browse/FAL-3527).
- Closes https://github.com/openedx/modular-learning/issues/120

## Testing instructions

- Turn on the `new_studio_mfe.use_tagging_taxonomy_list_page` flag on [studio](http://localhost:18000/admin/waffle/flag/).
- Run [taxonomy-sample-data](https://github.com/open-craft/taxonomy-sample-data) script to create sample data.
- On lms shell run `paver update_assets`. You need to clear the cache of your browser or use an incognito session
- Make sure you have the [frontend-app-course-authoring](https://github.com/openedx/frontend-app-course-authoring/) repo cloned and checkout the branch of this PR https://github.com/openedx/frontend-app-course-authoring/pull/654
- Go to a [new created unit](http://localhost:18010/container/block-v1:SampleTaxonomyOrg1+STC1+2023_1+type@vertical+block@aaf8b8eb86b54281aeeab12499d2cb0b) and check the tag list.
- Click on "Manage tags" and check that opens the drawer.
- Open the menu of a component and click on "Manage tags". Check that opens the drawer
- Turn **off** the `new_studio_mfe.use_tagging_taxonomy_list_page` flag on [studio](http://localhost:18000/admin/waffle/flag/).
- Return to the unit page and verify that the tag list and the "Manage tags" menu item not shown

## Additional info

- [Figma discussion on expanding the children of a taxonomy](https://www.figma.com/file/DmUC2e7BAoQsGDq4KgApPa?node-id=9834:20156&mode=design#604273107)

## TODO

- Wait https://github.com/openedx/modular-learning/issues/118 to connect the drawer with the "Manage tags" buttons.
